include/roboteam_ai/control/BBTrajectories/BBTrajectory1D.h

```diff
@@ -0,0 +1,124 @@
+//
+// Created by rolf on 15-01-20.
+//
+
+#ifndef RTT_BBTRAJECTORY1D_H
+#define RTT_BBTRAJECTORY1D_H
+
+/**
+ * @author rolfvdhulst
+ * @date 15 January 2020
+ * @tparam num : float or double
+ */
+template<class num>
+struct BBTrajectoryPart {
+  num tEnd;
+  num acc;
+  num startVel;
+  num startPos;
+};
+
+/**
+ * @author rolfvdhulst
+ * @date 15 January 2020
+ * @tparam num : float or double
+ */
+template<class num>
+struct PosVelAcc {
+  explicit PosVelAcc(num pos, num vel, num acc)
+          :
+          pos{pos},
+          vel{vel},
+          acc{acc} { };
+
+  num pos;
+  num vel;
+  num acc;
+};
+
+/**
+ * @author rolfvdhulst
+ * @date 15 January 2020
+ * @tparam num : float or double
+ */
+template<class num>
+class BBTrajectory1D {
+    public:
+        using part = BBTrajectoryPart<num>;
+        PosVelAcc<num> getValues(num t) const noexcept ;
+        num getPosition(num t) const noexcept;
+        num getVelocity(num t) const noexcept;
+        num getAcceleration(num t) const noexcept;
+        num getTotalTime() const noexcept ;
+        bool inLastPart(num t) const noexcept;
+        /**
+        * Generate a time-optimal trajectory given the parameters
+        * @param initialPos Current position
+        * @param initialVel Current velocity
+        * @param finalPos Final position
+        * @param maxVel Maximum allowed velocity (absolute)
+        * @param maxAcc maximum allowed acceleration/deceleration
+        */
+        void generateTrajectory(num initialPos, num initialVel, num finalPos, num maxVel, num maxAcc) noexcept ;
+        BBTrajectory1D(num initialPos, num initialVel,num finalPos, num maxVel, num maxAcc) noexcept;
+        BBTrajectory1D() noexcept = default;
```

Comment:
= default is already noexcept
```suggestion
        BBTrajectory1D() = default;
```